### PR TITLE
Fix more acceptance tests in slotted runtime

### PR DESCRIPTION
**Pull request by fickludd**
Most failures cause by `ExecutionContext.copyWith` which creates a new
`ExecutionContext` with the slot configuration of the old one. This wrong
as new `ExecutionContext`s have to use the slot configuration of the
current operator, because these could be different. Creating
`ExecutionContext`s with the wrong slot configuration typically gives
bug with the error msg "Ouch, no suitable slot for key ...".

Builds on #10645, first commit is https://github.com/neo4j/neo4j/commit/812db816523a774d5777b399d850464712f1a7b8.

### CODE REVIEW

File: community/cypher/interpreted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/interpreted/pipes/LoadCSVPipe.scala [hunk 1]
```diff
@@ -87,15 +88,22 @@ case class LoadCSVPipe(source: Pipe,
         //we need to make a copy here since someone may hold on this
         //reference, e.g. EagerPipe
 
-        context.copyWith(variable, VirtualValues.map(internalMap.copy.mapValues(v => if (v == null) Values.NO_VALUE else v).asJava))
+        val resultCopy = new util.HashMap[String, AnyValue](internalMap.size)
+        for (kv <- internalMap) {
```
Comment (pontusmelke) [inline community/cypher/interpreted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/interpreted/pipes/LoadCSVPipe.scala]:
```
for ( (key,value) <- internalMap ) {...
```

File: community/cypher/interpreted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/interpreted/pipes/NodeCountFromCountStorePipe.scala [hunk 2]
```diff
@@ -42,6 +41,8 @@ case class NodeCountFromCountStorePipe(ident: String, labels: List[Option[LazyLa
           count *= state.query.nodeCountByCountStore(NameId.WILDCARD)
       }
     }
-    Seq(baseContext.copyWith(ident, Values.longValue(count))).iterator
+
+    val baseContext = state.createOrGetInitialContext(executionContextFactory)
+    Seq(executionContextFactory.copyWith(baseContext, ident, Values.longValue(count))).iterator
```
Comment (pontusmelke) [inline community/cypher/interpreted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/interpreted/pipes/NodeCountFromCountStorePipe.scala]:
```
Iterator(executionContextFactory.copyWith(baseContext, ident, Values.longValue(count)))
```

File: enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/slotted/SlottedExecutionContext.scala [hunk 3]
```diff
@@ -191,42 +191,32 @@ case class SlottedExecutionContext(slots: SlotConfiguration) extends ExecutionCo
     this
   }
 
-  // This method is used instead of newWith1 from a ScopeExpression where the key in the new scope is overriding an existing key
-  // and it has to have the same name due to syntactic constraints. In this case we actually make a copy in order to not corrupt the existing slot.
   override def copyWith(key1: String, value1: AnyValue): ExecutionContext = {
+    // This method should throw like it siblings below, as soon as reduce is changed to not use it.
```
Comment (pontusmelke) [inline enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/slotted/SlottedExecutionContext.scala]: ```like its siblings below as soon...```

File: enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/slotted/SlottedExecutionContext.scala [hunk 4]
```diff
@@ -191,42 +191,32 @@ case class SlottedExecutionContext(slots: SlotConfiguration) extends ExecutionCo
...
   }
 
   override def copyWith(key1: String, value1: AnyValue, key2: String, value2: AnyValue): ExecutionContext = {
-    val newCopy = SlottedExecutionContext(slots)
-    copyTo(newCopy)
-    newCopy.setValue(key1, value1)
-    newCopy.setValue(key2, value2)
-    newCopy
+    throw new UnsupportedOperationException(
+      "Use ExecutionContextFactory.copyWith instead, to get the correct slot configuration"
```
Comment (pontusmelke) [inline enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/slotted/SlottedExecutionContext.scala]:
```
"Use ExecutionContextFactory.copyWith instead to get the correct slot configuration"
```

File: enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/slotted/SlottedExecutionContext.scala [hunk 5]
```diff
@@ -191,42 +191,32 @@ case class SlottedExecutionContext(slots: SlotConfiguration) extends ExecutionCo
...
                         key2: String, value2: AnyValue,
                         key3: String, value3: AnyValue): ExecutionContext = {
-    val newCopy = SlottedExecutionContext(slots)
-    copyTo(newCopy)
-    newCopy.setValue(key1, value1)
-    newCopy.setValue(key2, value2)
-    newCopy.setValue(key3, value3)
-    newCopy
+    throw new UnsupportedOperationException(
+      "Use ExecutionContextFactory.copyWith instead, to get the correct slot configuration"
```
Comment (pontusmelke) [inline enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/slotted/SlottedExecutionContext.scala]:
```
"Use ExecutionContextFactory.copyWith instead to get the correct slot configuration"
```

File: enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/slotted/SlottedExecutionContext.scala [hunk 6]
```diff
@@ -191,42 +191,32 @@ case class SlottedExecutionContext(slots: SlotConfiguration) extends ExecutionCo
...
   override def copyWith(newEntries: Seq[(String, AnyValue)]): ExecutionContext = {
-    val newCopy = SlottedExecutionContext(slots)
-    copyTo(newCopy)
-    for (t <- newEntries) {
-      val (key,value) = t
-      newCopy.setValue(key, value)
-    }
-    newCopy
+    throw new UnsupportedOperationException(
+      "Use ExecutionContextFactory.copyWith instead, to get the correct slot configuration"
```
Comment (pontusmelke) [inline enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/slotted/SlottedExecutionContext.scala]:
```
"Use ExecutionContextFactory.copyWith instead to get the correct slot configuration"
```

File: enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/slotted/SlottedQueryState.scala [hunk 7]
```diff
@@ -72,10 +72,42 @@ case class SlottedExecutionContextFactory(slots: SlotConfiguration) extends Exec
     newCtx
   }
 
+  override def copyWith(row: ExecutionContext, newEntries: Seq[(String, AnyValue)]): ExecutionContext = {
+    val newCopy = SlottedExecutionContext(slots)
+    row.copyTo(newCopy)
+    for (t <- newEntries) {
```
Comment (pontusmelke) [inline enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/slotted/SlottedQueryState.scala]:
```
for ((key,value) <- newEntries) {...
```

**Review by pontusmelke — CHANGES_REQUESTED** [review]
Nice, just had a couple of cosmetic changes.

File: enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/slotted/SlottedQueryState.scala (hunk 7 above)
Comment (fickludd) [reply enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/slotted/SlottedQueryState.scala]: Thanks, I was somehow under the impression that scalac didn't support that for some ambiguity reason. Maybe from some previous scala version? Much better now.

**Comment by fickludd** [comment]
ping @pontusmelke. I agreed.

**Review by pontusmelke — APPROVED** [review]

**Pull request merged**